### ISSUE TITLE
Export pattern as WAV file

### ISSUE THREAD

**Issue by grampalak** [hydrogen-music/hydrogen#1036]
Hydrogen version * :  1.0.1      
Operating system + version: Windows 8.1 Pro
Audio driver + version: Realtek integrated

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

< While Hydrogen will export a song as a wav file, the inability to do this with a selected pattern is disappointing.  Although the Project Menu dropdown seems to indicate that this is going to be possible (Export pattern as...), no format other than as a .h2 pattern is available.  I'd like to suggest this capability as a new feature request.  Otherwise, I love what Hydrogen allows me to do!  Thanks for your hard work.  :-)- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

*  H2 version : see 'Info - About', or in case you build Hydrogen from source use 'git rev-parse HEAD'
** if you are reporting a bug, please describe how the bug can be reproduced

**Comment by AkKrock** [comment]
As a suggestion, adding some info regarding Interpolation future (on export screen) could be useful.

**Comment by grampalak** [comment]
@AkKrock 
I'm not sure I understand "interpolation future". Do you mind adding some detail?  Thanks for your input.

**Comment by cme** [comment]
Should there be a "loopable" option? 

There are different use cases for exporting a pattern as a wave:
  - one-shot, plays once through (eg. for a break or for teaching a pattern)
  - looping the same pattern over and over

Using the existing export methods, by default a pattern won't be loopable (depending on what's in it). Consider something like a cymbal crash on a pickup beat like:

```
BD: |O---O---O---O---|
CC: |---------------X|
```

To make the pattern loopable, you'd want the start of the pattern to contain the sustain of the crash; but not when it'd be used as a one-shot.

**Comment by AkKrock** [comment]
> @AkKrock
> I'm not sure I understand "interpolation future". Do you mind adding some detail? Thanks for your input.

During export, there is an interpolation feature. Explanation of it is shared before in a webpage reference. But I think a quick info could be useful for this feature in export screen. Just a simple suggestion, don't mind. =)

**Comment by oddtime** [comment]
> To make the pattern loopable, you'd want the start of the pattern to contain the sustain of the crash; but not when it'd be used as a one-shot.

Maybe a toggle button to include or not the very last sustain? I mean working in both modes _one-shot_ and _looping_
After all _one-shot_ mode is just a loop with 1 only repetition, so there could be a `QSpinBox` in the export dialog to select `nRepetitions`.

**Comment by oddtime** [comment]
> During export, there is an interpolation feature. Explanation of it is shared before in a webpage reference. But I think a quick info could be useful for this feature in export screen. Just a simple suggestion, don't mind. =)

I agree. The interpolation is used (at least) when samples are resampled at different rate frequency for the "doppler like" note pitch shift.
